src/werkzeug/debug/tbtools.py

```diff
@@ -466,102 +436,34 @@ def render_line(line, cls):
-        except IndexError:
-            pass
-
-        return lines
+    def get_context_lines(self, context=5):
+        lines = linecache.getlines(self.filename)
+        start_idx = max(0, self.lineno - context)
+        stop_idx = min(len(lines), self.lineno + context)
+
+        before = lines[start_idx : self.lineno]
+        after = lines[self.lineno + 1 : stop_idx]
+
+        return before, self.line, after
+
+    def render_text(self):
+        return (
+            f'  File "{self.filename}", line {self.lineno}, in {self.name}\n'
+            f"    {self.line}"
+        )
 
     def eval(self, code, mode="single"):
         """Evaluate code in the context of the frame."""
         if isinstance(code, str):
             code = compile(code, "<interactive>", mode)
         return eval(code, self.globals, self.locals)
 
-    @cached_property
-    def sourcelines(self):
-        """The sourcecode of the file as list of strings."""
-        # get sourcecode from loader or file
-        source = None
-        if self.loader is not None:
-            try:
-                if hasattr(self.loader, "get_source"):
-                    source = self.loader.get_source(self.module)
-                elif hasattr(self.loader, "get_source_by_code"):
-                    source = self.loader.get_source_by_code(self.code)
-            except Exception:
-                # we munch the exception so that we don't cause troubles
-                # if the loader is broken.
-                pass
-
-        if source is None:
-            try:
-                with open(self.filename, mode="rb") as f:
-                    source = f.read()
-            except OSError:
-                return []
-
-        # already str?  return right away
-        if isinstance(source, str):
-            return source.splitlines()
-
-        charset = "utf-8"
-        if source.startswith(UTF8_COOKIE):
-            source = source[3:]
-        else:
-            for idx, match in enumerate(_line_re.finditer(source)):
-                match = _coding_re.search(match.group())
-                if match is not None:
-                    charset = match.group(1)
-                    break
-                if idx > 1:
-                    break
-
-        # on broken cookies we fall back to utf-8 too
-        charset = _to_str(charset)
-        try:
-            codecs.lookup(charset)
-        except LookupError:
-            charset = "utf-8"
-
-        return source.decode(charset, "replace").splitlines()
-
-    def get_context_lines(self, context=5):
-        before = self.sourcelines[self.lineno - context - 1 : self.lineno - 1]
-        past = self.sourcelines[self.lineno : self.lineno + context]
-        return (before, self.current_line, past)
-
-    @property
-    def current_line(self):
-        try:
-            return self.sourcelines[self.lineno - 1]
-        except IndexError:
-            return ""
-
-    @cached_property
-    def console(self):
```

Comment:
@teymour-aldridge just checking this was meant to be deleted?

Reply:
I'm not sure. Please take anything I've done here with a degree of scepticism – I'm 100% sure about the code I've modified.